test/e2e/provisioner/gce.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2019 The KubeOne Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provisioner
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/kubermatic/kubeone/test/e2e/testutil"
+)
+
+// GCEProvisioner contains implementation of GCE provisioner interface
+type GCEProvisioner struct {
+	testPath  string
+	terraform *terraform
+
+	init bool
+}
+
+// NewGCEProvisioner creates and initialize GCE provisioner
+func NewGCEProvisioner(creds func() error, testPath, identifier string) (*GCEProvisioner, error) {
+	if err := creds(); err != nil {
+		return nil, errors.Wrap(err, "unable to validate credentials")
+	}
+
+	terraform := &terraform{
+		terraformDir: "../../examples/terraform/gce/",
+		identifier:   identifier,
+	}
+
+	return &GCEProvisioner{
+		terraform: terraform,
+		testPath:  testPath,
+
+		init: false,
+	}, nil
+}
+
+// Provision provisions a cluster using Terraform
+func (p *GCEProvisioner) Provision() (string, error) {
+	args := []string{}
+	if !p.init {
+		args = []string{"-var", "control_plane_target_pool_members_count=1"}
```

Comment:
is new type GCEProvisioner needed only for this line?

Reply:
/shrug

Basically. Another way was to extend DefaultProvisioner to take provider and add an init field, and then if provider GCE and if not init, add args and set init to true.

I don't have strong opinions, but provisioner is already small and mostly about that, so I went with a new one over extending default one.

Reply:
The GCEProvisioner is removed in favor of the proposed approach.

/unshrug